### PR TITLE
Add command to get screen density from ADB.

### PR DESCRIPTION
**Pull request by fuzzybinary**

### CODE REVIEW

File: lib/tools/adb-commands.js [hunk 1]
```diff
@@ -1338,6 +1338,21 @@ methods.getScreenSize = async function () {
   return null;
 };
 
+/**
+ * Get the current screen density in dpi
+ *
+ * @return {string} Device screen density as a string or _null_ if it
```
Comment (mykola-mokhnach) [inline lib/tools/adb-commands.js]: shouldn't it be a float number?

File: lib/tools/adb-commands.js [hunk 2]
```diff
@@ -1338,6 +1338,21 @@ methods.getScreenSize = async function () {
...
+ *                  cannot be determined
+ */
+methods.getScreenDensity = async function () {
+  let stdout = await this.shell(['wm', 'density']);
+  let size = new RegExp(/Physical density: ([^\r?\n]+)*/g).exec(stdout);
+  if (size && size.length >= 2) {
+    return size[1].trim();
+  }
+  return null;
```
Comment (mykola-mokhnach) [inline lib/tools/adb-commands.js]: I'd rather throw an exception instead

File: lib/tools/adb-commands.js (hunk 1 above)
Comment (fuzzybinary) [reply lib/tools/adb-commands.js]: I did think about that... maybe?

File: lib/tools/adb-commands.js (hunk 2 above)
Comment (fuzzybinary) [reply lib/tools/adb-commands.js]: I'm matching previous functionality there from getScreenSize.  Since I intend on dropping this into the caps of the device for reading, I figure null makes sense?

**Comment by fuzzybinary** [comment]
Okay, I've changed this to return a number instead of a string.

File: lib/tools/adb-commands.js [hunk 3]
```diff
@@ -1338,6 +1338,22 @@ methods.getScreenSize = async function () {
...
+ * Get the current screen density in dpi
+ *
+ * @return {string} Device screen density as a number or _null_ if it
+ *                  cannot be determined
+ */
+methods.getScreenDensity = async function () {
+  let stdout = await this.shell(['wm', 'density']);
+  let density = new RegExp(/Physical density: ([^\r?\n]+)*/g).exec(stdout);
+  if (density && density.length >= 2) {
+    let densityNumber = Number(density[1].trim());
```
Comment (mykola-mokhnach) [inline lib/tools/adb-commands.js]: `= parseInt(..., 10)`

File: lib/tools/adb-commands.js [hunk 4]
```diff
@@ -1338,6 +1338,22 @@ methods.getScreenSize = async function () {
...
+    return Number.isNaN(densityNumber) ? null : densityNumber;
```
Comment (mykola-mokhnach) [inline lib/tools/adb-commands.js]: just `isNaN`

File: lib/tools/adb-commands.js [hunk 5]
```diff
@@ -1338,6 +1338,22 @@ methods.getScreenSize = async function () {
   return null;
 };
 
+/**
+ * Get the current screen density in dpi
+ *
+ * @return {string} Device screen density as a number or _null_ if it
```
Comment (mykola-mokhnach) [inline lib/tools/adb-commands.js]: please update the docstring :`{?number}`

File: test/unit/adb-commands-specs.js [hunk 6]
```diff
@@ -792,6 +793,13 @@ describe('adb commands', () => {
       await adb.getScreenSize();
       mocks.adb.verify();
     });
+    it('should get device screen density', async () => {
+      mocks.adb.expects("shell")
+          .once().withExactArgs(['wm', 'density'])
+          .returns(screenDensity);
+      await adb.getScreenDensity();
```
Comment (mykola-mokhnach) [inline test/unit/adb-commands-specs.js]: Plase add the verification of the returned result. Also it would be good to have a test, which verifies `null` output case

File: package-lock.json [hunk 7]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "appium-adb",
-  "version": "3.0.0",
+  "version": "5.0.0",
```
Comment (mykola-mokhnach) [inline package-lock.json]: Please revert this

Comment (mykola-mokhnach) [reply package-lock.json]: also I'm not sure why `test/fixtures/selendroid-test-app.apk` is present in the PR

Comment (fuzzybinary) [reply package-lock.json]: Me either actually. It's in the repo, but likely it should probably be .gitignored?

**Comment by imurchie** [comment]
Published in `appium-adb@5.1.0`.
